flow/keys/save/kms/kms.go

```diff
@@ -39,46 +41,57 @@ type Config struct {
 	Overwrite  bool   `flag:"overwrite" info:"bool indicating if we should overwrite an existing config with the same name in the config file"`
 }
 
-var conf Config
+var flags Flags
 
 var Cmd = &cobra.Command{
 	Use:     "kms",
 	Short:   "Save a KMS key to the config file",
 	Example: "flow keys save kms --name test --address 8c5303eaa26202d6 --sigalgo ECDSA_secp256k1 --hashalgo SHA2_256 --index 0 --context 'KMS_RESOURCE_ID'",
 	Run: func(cmd *cobra.Command, args []string) {
-		projectConf := cli.LoadConfig()
+		project := cli.LoadProject()
 
-		_, accountExists := projectConf.Accounts[conf.Name]
-		if accountExists && !conf.Overwrite {
-			cli.Exitf(1, "%s already exists in the config, and overwrite is false", conf.Name)
+		account := project.GetAccountByName(flags.Name)
+		if account != nil && !flags.Overwrite {
+			cli.Exitf(1, "%s already exists in the config, and overwrite is false", flags.Name)
 		}
 
-		keyContext, err := cli.KeyContextFromKMSResourceID(conf.KeyContext)
+		keyContext, err := keys.KeyContextFromKMSResourceID(flags.KeyContext)
 		if err != nil {
-			cli.Exitf(1, "key context could not be parsed %s", conf.KeyContext)
+			cli.Exitf(1, "key context could not be parsed %s", flags.KeyContext)
 		}
-		// Populate account
-		account := &cli.Account{
-			KeyType:    cli.KeyTypeKMS,
-			Address:    flow.HexToAddress(conf.Address),
-			SigAlgo:    crypto.StringToSignatureAlgorithm(conf.SigAlgo),
-			HashAlgo:   crypto.StringToHashAlgorithm(conf.HashAlgo),
-			KeyIndex:   conf.KeyIndex,
-			KeyContext: keyContext,
+
+		address := flow.HexToAddress(flags.Address)
+		if !address.IsValid(flow.Emulator) { // TODO: don't hardcode this
+			cli.Exitf(1, "key address not valid")
 		}
 
+		keys := []config.AccountKey{{
+			Type:     config.KeyTypeKMS,
+			Index:    flags.KeyIndex,
+			SigAlgo:  crypto.StringToSignatureAlgorithm(flags.SigAlgo),
+			HashAlgo: crypto.StringToHashAlgorithm(flags.HashAlgo),
+			Context:  keyContext,
+		}}
+
+		account, err := cli.AccountFromConfig(
+			config.Account{
+				Name:    flags.Name,
+				Address: address,
+				ChainID: flow.Emulator, // TODO: don't hardcode this
```

Comment:
I'm starting to question the value of having `ChainID` as part of the account config...

Reply:
Let's please have a conversation on this topic because I also think it's really non-consistent currently (it's defined on networks, somewhere on accounts etc...)

Reply:
I found this confusing as well, when trying to set up the config.